RecoTracker/DeDx/plugins/DeDxHitCalibrator.cc

```diff
@@ -0,0 +1,426 @@
+  if (subdet == TECThin && thickness > 400e-4)
+    return TECThick;
+  return subdet;
+}
+
+/*****************************************************************************/
+float DeDxHitCalibrator::correctEnergy(const float& energy, const ChipId& detId) {
+  const auto& g = dedxCalib_->gain().find(detId);
+  if (applyGain_ && g != dedxCalib_->gain().end())
+    return energy * g->second;
+  return energy;
+}
+
+/*****************************************************************************/
+double DeDxHitCalibrator::getChi2(const std::vector<double>& x,
+                                  const std::vector<std::pair<double, int> >& b,
+                                  const double& coupling,
+                                  const double& iSigma) {
+  const auto& npar = b.size();
+  std::vector<double> y(npar, 0.);
+  for (size_t i = 0; i < npar; i++) {
+    const auto dx = coupling * x[i];
+    if (i >= 1)
+      y[i - 1] += dx;
+    y[i] += x[i] - 2 * dx;
+    if (i + 1 < npar)
+      y[i + 1] += dx;
+  }
+  double chi2(0.);
+  for (size_t i = 0; i < npar; i++) {
+    auto q = (b[i].first - y[i]) * iSigma;
+    if (b[i].second == kIsNormal)
+      chi2 += q * q;
+    if (b[i].second == kIsBelow) {
+      q -= 2;
+      if (q < 0)
+        chi2 += 0.5 * q * q;
+    }
+    if (b[i].second == kIsOver) {
+      q += 2;
+      if (q > 0)
+        chi2 += 0.5 * q * q;
+    }
+    // Penalty for negatives
+    if (x[i] < 0) {
+      q = x[i] * iSigma;
+      chi2 += q * q;
+    }
+  }
+  return chi2;
+}
+
+/*****************************************************************************/
+void DeDxHitCalibrator::getAlphaBeta(const std::vector<double>& x,
+                                     const std::vector<std::pair<double, int> >& b,
+                                     CLHEP::HepMatrix& alpha,
+                                     CLHEP::HepVector& beta,
+                                     const std::vector<bool>& isFix,
+                                     const double& coupling,
+                                     const double& iSigma) {
+  const auto& npar = b.size();
+  const auto a0 = coupling * iSigma;
+  const auto a1 = (1 - 2 * coupling) * iSigma;
+  const auto iSigmaSqr = iSigma * iSigma;
+  const auto a00 = coupling * coupling * iSigmaSqr;
+  const auto a11 = (1 - 2 * coupling) * (1 - 2 * coupling) * iSigmaSqr;
+  const auto a01 = coupling * (1 - 2 * coupling) * iSigmaSqr;
+  std::vector<double> y(npar, 0.);
+  for (size_t i = 0; i < npar; i++)
+    if (!isFix[i]) {
+      const auto dx = coupling * x[i];
+      if (i >= 1)
+        y[i - 1] += dx;
+      y[i] += x[i] - 2 * dx;
+      if (i + 1 < npar)
+        y[i + 1] += dx;
+    }
+  for (size_t i = 0; i < npar; i++) {
+    auto q = (y[i] - b[i].first) * iSigma;
+    int f(0);
+    if (b[i].second == kIsNormal)
+      f = 2;
+    if (b[i].second == kIsBelow) {
+      q += 2;
+      if (q > 0)
+        f = 1;
+    }
+    if (b[i].second == kIsOver) {
```

Comment:
```suggestion
    else if (b[i].second == kIsOver) {
```

Reply:
Addressed in 05c2dc85edef1e8e3a387cd8694941aa4c4e24bd